snapd-glib/requests/snapd-json.c

```diff
@@ -101,6 +103,30 @@ _snapd_json_get_object (JsonObject *object, const gchar *name)
         return NULL;
 }
 
+const GPtrArray*
```

Comment:
If the links can only be strings, then this should return GStrv - it's easier to use than GPtrArray in that case.

Reply:
See tracks for an example of how this is done.

Reply:
The links are lists, not strings only anymore as I implemented in [snapcraft also](https://github.com/canonical/snapcraft/pull/4652).

Is the current implementation not okay?

Reply:
GPtrArray would be a good choice if snapd_json_get_links returned a list of objects. But each link is just a string, to GStrv is a simpler container to use.

See how the tracks are processed for a similar case:
```
    g_autoptr(JsonArray) tracks =_snapd_json_get_array (object, "tracks");
    g_autoptr(GPtrArray) track_array = g_ptr_array_new ();
    for (guint i = 0; i < json_array_get_length (tracks); i++) {
        JsonNode *node = json_array_get_element (tracks, i);
        if (json_node_get_value_type (node) != G_TYPE_STRING) {
            g_set_error (error, SNAPD_ERROR, SNAPD_ERROR_READ_FAILED, "Unexpected track type");
            return NULL;
        }
        g_ptr_array_add (track_array, (gpointer) json_node_get_string (node));
    }
    g_ptr_array_add (track_array, NULL);
```